### PR TITLE
Improve P2P transaction handling

### PR DESCRIPTION
**Pull request by roman-khimov**
cpu_four_(30_wrk,_64K_pool)
cpu_seven_(30_wrk,_64K_pool)
mem_four_(30_wrk,_64K_pool)
mem_seven_(30_wrk,_64K_pool)
ms_per_block_four_(30_wrk,_64K_pool)
ms_per_block_seven_(30_wrk,_64K_pool)
tpb_four_(30_wrk,_64K_pool)
tpb_seven_(30_wrk,_64K_pool)
tps_four_(30_wrk,_64K_pool)
netw
tps_seven_(30_wrk,_64K_pool)

### CODE REVIEW

File: pkg/network/server.go [hunk 1]
```diff
@@ -1042,19 +1049,39 @@ func (s *Server) handleTxCmd(tx *transaction.Transaction) error {
...
+		}
 	}
-	if s.verifyAndPoolTX(tx) == nil {
-		s.broadcastTX(tx, nil)
+drainloop:
+	for {
+		select {
+		case <-s.txin:
+		default:
+			break drainloop
```
Comment (AnnaShaleva) [inline pkg/network/server.go:1082]: And close `txin` after that?

**Review by AnnaShaleva — COMMENTED** [review]
Tests need to be fixed.

**Comment by roman-khimov** [comment]
> Tests need to be fixed.

Yep, already done.

File: pkg/network/server.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/network/server.go:1082]: `txHandlerLoop()` can't do that, we have a number of them. We don't close `s.transactions` also, GC will handle it, although we only do this when server shuts down and application exits.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**